server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -181,6 +184,11 @@ public NodeStats(StreamInput in) throws IOException {
         } else {
             fileCacheStats = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_2_7_0) && FeatureFlags.isEnabled(FeatureFlags.REPLICATION_TYPE)) {
```

Comment:
Hey great questions.
1. Yeah I originally had this as 3.0.0 without a ff check, but this will break bwc tests until backported.  I'm not a fan of merging anything with a breaking test, so threw that out.
I changed this to 2.7.0 because thats the next minor that this will be backported to - this still breaks bwc tests until the change is backported to that branch.   I added the FF check because that fixes the test while also making sense to not return anything SR related if the ff is off.  I'd prefer we still include the FF where required until we know we are 100% going with that release, and remove/flip the fflag default in one commit.

2. Yeah this is a risk that we accepted with our flags when they were introduced. We don't do any node-node validation of flags, requiring the need to be intentional about flipping them on.

Reply:
Thanks @mch2 for the update.
1. Yes, IMO using `3.0.0` version check should work. But, agree on keeping the existing check as it is.
2. Agree. This will break lot many things :D